packages/generator/src/generators/app-generator.ts

```diff
@@ -38,6 +41,22 @@ export class AppGenerator extends Generator<AppGeneratorOptions> {
     }
   }
 
+  getPromptRunner(): PromptRunner {
+    const formChoices: Array<{name: AppGeneratorPromptOptions["form"]; message?: string}> = [
+      {name: "React Final Form", message: "React Final Form (recommended)"},
+      {name: "React Hook Form"},
+      {name: "Formik"},
+    ]
+
+    const formPrompt: Prompt = {
+      name: "form",
+      message: "Pick a form library (you can switch to something else later if you want)",
+      choices: formChoices,
+    }
+
+    return new PromptRunner([formPrompt])
+  }
```

Comment:
So I'm curious why we'd take this approach of having to return a `PromptRunner` - it seems like we could just make this a class property, right? 

```
// prompts: Record<string, PromptConfig>
this.prompts = {
  formChoices: formPrompt
} as const;
```

That should allow us to safely type access to `this.promptResults` without having to introduce a new method to the generator interface.

Reply:
I do not know how I can make the typing work for promptResult here to be honest. If I do it like you suggest I can run the prompts like this

    for (let prompt of this.prompts) {
      const promptResult = await this.enquirer.prompt({
        ...prompt,
        type: "select",
      })

      this.promptResults = {
        ...this.promptResults,
        [prompt.name]: promptResult
      }
    }

But this gives me TypeError for this.promptResults

_Error:(298, 7) TS2322: Type '(S & { [x: string]: object; }) | {}' is not assignable to type 'S | undefined'.
  Type '{}' is not assignable to type 'S'.
    '{}' is assignable to the constraint of type 'S', but 'S' could be instantiated with a different subtype of constraint 'object'._

This is the reason why I went with the Runner.

Reply:
@engelkes-finstreet I'm not totally following here, but you can definitely make it work with `as any`:
```ts
  this.promptResults = {
    ...this.promptResults,
    [prompt.name]: promptResult
  } as any
```

Do that to get past this. Then I can check the code and try to get a proper fix or it.